test/sensitivities/functional/functional.jl

```diff
@@ -140,15 +123,15 @@ using DiffRules: diffrule, hasdiffrule
             z2_ = broadcast(/, x_, y_)
             @test unbox(z_) == x ./ y
             @test ∇(z_, oneslike(unbox(z_)))[x_] == ∇(z2_, oneslike(unbox(z2_)))[x_]
-            @test ∇(z_, oneslike(unbox(z_)))[y_] == ∇(z2_, oneslike(unbox(z2_)))[y_]
+            @test ∇(z_, oneslike(unbox(z_)))[y_] ≈ ∇(z2_, oneslike(unbox(z2_)))[y_]
```

Comment:
Should this not remain as `==`?

Reply:
I don't think it has to.
`x ./ 5.0` and `x / 5.0` are computed differently and so can accumulate different floating point errors.

I think internally Zygote was implementing the later using the former, where as ChainRules doesn't.

Chainrules does 
https://github.com/JuliaDiff/ChainRules.jl/blob/13b11bd02a6713bb614a1974f2b5aa4c7a2087cd/src/rulesets/Base/arraymath.jl#L127
which uses 
`-dot(Ȳ, Y)/b` 

Where as Nabla used to do
https://github.com/invenia/Nabla.jl/blob/4cadc87677fb1187354999dcf93bd528f45f85d0/src/sensitivities/functional/functional.jl#L131
where it did directly delegate it to broadcasting

and boils eventually down to
`sum(Ȳ .* .-Y./b)` 

and while mathematically they are the same,
they do slightly diverge eventually

```
julia> x = 10randn(10_000); y=10randn(10_000);

julia> sum(x.*y) == dot(x,y)
false

julia> sum(x.*y) - dot(x,y)
2.7284841053187847e-12
``

Reply:
Julia Slack thread for this, although it'll get lost one day,

> https://julialang.slack.com/archives/C677F4ABD/p1603462940170200